pkg/types/reference.go

```diff
@@ -0,0 +1,100 @@
+package types
+
+import (
+	"fmt"
+	"go/token"
+	"go/types"
+	"reflect"
+	"strings"
+
+	"github.com/crossplane-contrib/terrajet/pkg/terraform/resource"
+
+	"github.com/pkg/errors"
+	"golang.org/x/tools/go/packages"
+
+	"github.com/crossplane-contrib/terrajet/pkg/comments"
+)
+
+const (
+	// PackagePathXPCommonAPIs is the go path for the Crossplane Runtime package
+	// with common APIs
+	PackagePathXPCommonAPIs = "github.com/crossplane/crossplane-runtime/apis/common/v1"
+
+	loadMode = packages.NeedName | packages.NeedImports | packages.NeedTypes
+)
+
+var typeXPRef types.Type
+var typeXPSelector types.Type
+var commentOptional *comments.Comment
```

Comment:
nitpick
```suggestion
var TypeReferenceField types.Type
var TypeSelectorField types.Type
var CommentOptional *comments.Comment
```

Reply:
Do you want to make these public, even only used in the same package? Why?

Reply:
I guess it's my personal tendency, there isn't a solid reason to make them public.